### DISCUSSION TITLE
Problems when configuring cypress-cloud

### DISCUSSION THREAD

**Discussion by luischre** [currents-dev/cypress-currents-reporter#140] [Q&A]
We are currently trying to migrate from cypress 9 to 12 and am running into the following error trying to use cypress-cloud:

>Platform: linux-x64 (Debian - 11.6)
Cypress Version: 12.13.0
Cypress stderr:
 [FAILED] Cypress failed to start.
[FAILED] 
[FAILED] This may be due to a missing library or dependency. https://on.cypress.io/required-dependencies
[FAILED] 
[FAILED] Please refer to the error below for more details.
[FAILED] 
[FAILED] ----------
[FAILED] 
[FAILED] [352:0609/143502.127077:ERROR:node_bindings.cc(279)] Most NODE_OPTIONs are not supported in packaged apps. See documentation for more details.
[FAILED] 
[FAILED] ----------
[FAILED] 
[FAILED] Platform: linux-x64 (Debian - 11.6)
[FAILED] Cypress Version: 12.13.0
 ERROR  Unable to resolve cypress configuration
>- make sure that 'cypress-cloud/plugin' is installed
>- report the issue together with cypress stdout and stderr`

Our setup is as follows:
- We create custom docker images with the provided factory dockerfile from cypress
- These docker images are then used within a kubernetes cluster
- We run 10 of these pods in parallel
- Within each pod we check out the repository and run `npm install`

Now it's starting to get weird:
- 5 out of those 10 pods succeed to start test runners
- The other 5 fail with the above mentioned error

Now it's getting weirder:
- Also within the pods that manage to survive they get almost the same error sometimes but only after test runner started which then afterwards crashes but the pod however survives and picks up the next spec file:

>Cypress stdout:
>The Test Runner unexpectedly exited via a exit event with signal SIGTRAP
>Platform: linux-x64 (Debian - 11.6)
Cypress Version: 12.13.0
Cypress stderr:
 [783:0609/063733.469610:ERROR:node_bindings.cc(279)] Most NODE_OPTIONs are not supported in packaged apps. See documentation for more details.
 ERROR  Unable to resolve cypress configuration
>- make sure that 'cypress-cloud/plugin' is installed
>- report the issue together with cypress stdout and stderr

Has anyone experienced similar issues or has any idea why this could be happening?

**Comment by agoldis** [comment]
@luischre Please share more details about your configuration. Follow the process defined in https://github.com/currents-dev/cypress-cloud/issues/new?assignees=&labels=bug&projects=&template=1.bug.yaml

**Reply by luischre** [reply]
Hi @agoldis ,
thanks for your answer, sorry for the delay, just got back on the topic.

Environment information (I've shrunk it down a bit):
<details>

> + npx envinfo --system --binaries --browsers --npmPackages --duplicates --npmGlobalPackages
> npm WARN exec The following package was not found and will be installed: envinfo@7.10.0
> 
>   System:
>     OS: Linux 3.10 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
>     CPU: (4) x64 AMD EPYC 7502 32-Core Processor
>     Memory: 20.24 GB / 30.52 GB
>     Container: Yes
>     Shell: 5.1.4 - /bin/bash
>   Binaries:
>     Node: 18.16.1 - /usr/local/bin/node
>     npm: 9.5.1 - /usr/local/bin/npm
>   Browsers:
>     Chrome: 114.0.5735.106
>   npmPackages:
>     ...
>     @cypress/angular:  0.0.0-development 
>     @cypress/browserify-preprocessor: ^3.0.2 => 3.0.2 
>     @cypress/mount-utils:  0.0.0-development 
>     @cypress/react:  0.0.0-development 
>     @cypress/react18:  0.0.0-development 
>     @cypress/skip-test: ^2.6.1 => 2.6.1 
>     @cypress/svelte:  0.0.0-development 
>     @cypress/vue:  0.0.0-development 
>     @cypress/vue2:  0.0.0-development 
>     cypress: ^12.13.0 => 12.13.0 
>     cypress-cloud: ^1.7.4 => 1.7.4 
>     cypress-multi-reporters: ^1.6.3 => 1.6.3 
>     cypress-on-fix: ^1.0.2 => 1.0.2 
>     cypress-pipe: ^2.0.0 => 2.0.0 
>     cypress-slack-notify: ^1.14.0 => 1.14.0 
>     cypress-terminal-report: ^3.5.2 => 3.5.2 
>     cypress-wait-until: ^1.7.1 => 1.7.2 
>     ...
>   npmGlobalPackages:
>     corepack: 0.17.0
>     cypress-cloud: 1.8.2
>     cypress: 12.13.0
>     npm: 9.5.1
>     typescript: 5.1.6

</details>

I've also enabled all debug logs and found one interesting thing. My docker image contains cypress version 12.13.0 which is installed globally. However from the debug logs I can see the following:

> 2023-07-03T09:44:37.699Z currents:config resolving cypress config
> 2023-07-03T09:44:37.699Z currents:boot booting cypress...
> 2023-07-03T09:44:37.712Z currents:boot cypress executable location: /usr/local/lib/node_modules/cypress-cloud/node_modules/cypress/bin/cypress

As I have cypress installed globally why isn't it just using `cypress run` instead of `/usr/local/lib/node_modules/cypress-cloud/node_modules/cypress/bin/cypress run...`. This looks somehow weird to me.

Also not sure if it's coming from cypress cloud but for some reason my cypress cache actually also contains the latests cypress version (12.16.0) once the run is started so my guess is that it's somehow dragged in when installing some more packages, could this be when installing cypress-cloud?

It's actually trying to execute 12.16.0 and not the desired 12.13.0 and also their permission are different which is currently my main suspect.

**Reply by agoldis** [reply]
cypress-cloud has `cypress` as a peer dependency, and I think npm installs peer deps by default.

As for the error itself - something is wrong with how you setup cypress-cloud. Try following this guide: https://currents.dev/readme/integration-with-cypress/troubleshooting

Also, if you have more plugins installed, please refer to this guide: https://github.com/currents-dev/cypress-cloud#setup-with-existing-plugins

**Comment by luischre** [comment]
Hi @agoldis ,
thanks for your feedback! I haven't figured it out yet but at least narrowed it down to either cypress or out kubernetes setup, so cypress-cloud is out of the equation.
Just as some feedback, I would reconsider the error message that is thrown when using cypress-cloud as at least to me my first impression was that something was wrong with my cypress-cloud configuration and in the end it baiscally takes the blame for issues caused by other things.
